olp-cpp-sdk-core/src/utils/Dir.cpp

```diff
@@ -445,15 +445,14 @@ uint64_t Dir::Size(const std::string& path, FilterFunction filter_fn) {
   struct dirent* ent = nullptr;
 
   if ((dir = opendir(path.c_str())) != nullptr) {
-    int fd = dirfd(dir);
-
     while ((ent = readdir(dir)) != nullptr) {
+      std::string ent_path = path + "/" + ent->d_name;
 #ifdef __APPLE__
       struct stat sb;
-      if (fstatat(fd, ent->d_name, &sb, AT_SYMLINK_NOFOLLOW) == 0) {
+      if (lstat(ent_path.c_str(), &sb) == 0) {
```

Comment:
Have you tried to use lstat64? Or even stat64 is absent on macOS?

Reply:
lstat64() appeared in 10.5 and was deprecated in 10.6.
Please take a look at:
https://trac.osgeo.org/gdal/ticket/5780